pkg/log/log.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2018 Oracle and/or its affiliates. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"flag"
+	"fmt"
+	"sync"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var (
+	lvlString     = "info"
+	logJSON       = false
+	logfilePath   = ""
+	config        *zap.Config
+	mu            sync.Mutex
+	glogVerbosity string
+)
+
+func init() {
+	flag.StringVar(&lvlString, "log-level", lvlString, "Adjusts the level of the logs that will be omitted.")
```

Comment:
This can be cleaned up. 

```go
var (
	lvl     = zapcore.InfoLevel
	logJSON = true
	config  *zap.Config
	mu      sync.Mutex
)

func init() {
	flag.Var(&lvl, "log-level", "Adjusts the level of the logs that will be omitted.")
	flag.BoolVar(&logJSON, "log-json", logJSON, "Log in json format.")
}

....

config.Level.SetLevel(lvl)
```

Reply:
`zapcore.Level` is a `int8` though. Doesn't that required `--log-level=0` vs `--log-level=info`?

Reply:
That's why it's a `Var`. You'd pass strings. The level implements the flag interface https://github.com/uber-go/zap/blob/master/zapcore/level.go

Reply:
TIL there's a flag interface. Cheers 👍